sql/hive/src/test/scala/org/apache/spark/sql/hive/MetastoreDataSourcesSuite.scala

```diff
@@ -1167,16 +1161,15 @@ class MetastoreDataSourcesSuite extends QueryTest with SQLTestUtils with TestHiv
 
   test("create a temp view using hive") {
     val tableName = "tab1"
-    withTable (tableName) {
-      val e = intercept[ClassNotFoundException] {
-        sql(
-          s"""
-             |CREATE TEMPORARY VIEW $tableName
-             |(col1 int)
-             |USING hive
-           """.stripMargin)
-      }.getMessage
-      assert(e.contains("Failed to find data source: hive"))
+    withTable(tableName) {
+      sql(
+        s"""
+           |CREATE TEMPORARY VIEW $tableName
+           |(col1 int)
+           |USING hive
+         """.stripMargin)
+      val tempView = spark.sessionState.catalog.getTempView(tableName)
+      assert(tempView.isDefined, "create a temp view using hive should success")
```

Comment:
hmmm, it's not expected. let's add a check in `CreateTempViewUsing`, and throw exception for hive provider, e.g. `if (DDLUtils.isHiveTable(t)) throw ...`